connect_ext_test.go

```diff
@@ -714,3 +747,28 @@ func (p pingServer) CumSum(
 		}
 	}
 }
+
+type deflateReader struct {
+	r io.ReadCloser
+}
+
+func newDeflateReader(r io.Reader) *deflateReader {
+	return &deflateReader{r: flate.NewReader(r)}
+}
+
+func (d *deflateReader) Read(p []byte) (int, error) {
+	return d.r.Read(p)
+}
+
+func (d *deflateReader) Close() error {
+	return d.r.Close()
+}
+
+func (d *deflateReader) Reset(reader io.Reader) error {
+	if resetter, ok := d.r.(flate.Resetter); ok {
```

Comment:
Unfortunately `flate.NewReader` returns an `io.ReadCloser` instead of a concrete type. This type assertion should always succeed though.